homeassistant/components/alert/__init__.py

```diff
@@ -53,141 +58,206 @@
-        vol.Optional(CONF_ALERT_MESSAGE): cv.template,
-        vol.Optional(CONF_DONE_MESSAGE): cv.template,
-        vol.Optional(CONF_TITLE): cv.template,
-        vol.Optional(CONF_DATA): dict,
-        vol.Optional(CONF_NOTIFIERS, default=list): vol.All(
-            cv.ensure_list, [cv.string]
-        ),
-    }
-)
+STORAGE_KEY = DOMAIN
+STORAGE_VERSION = 1
+STORAGE_VERSION_MINOR = 0
+
+STORAGE_FIELDS = {
+    vol.Required(CONF_NAME): cv.string,
+    vol.Required(CONF_ENTITY_ID): cv.entity_id,
+    vol.Optional(CONF_STATE, default=STATE_ON): cv.string,
+    vol.Required(CONF_REPEAT): vol.All(
+        cv.ensure_list,
+        [vol.Coerce(float)],
+        # Minimum delay is 1 second = 0.016 minutes
+        [vol.Range(min=0.016)],
+    ),
+    vol.Optional(CONF_CAN_ACK, default=DEFAULT_CAN_ACK): cv.boolean,
+    vol.Optional(CONF_SKIP_FIRST, default=DEFAULT_SKIP_FIRST): cv.boolean,
+    vol.Optional(CONF_ALERT_MESSAGE): cv.template,
+    vol.Optional(CONF_DONE_MESSAGE): cv.template,
+    vol.Optional(CONF_TITLE): cv.template,
+    vol.Optional(CONF_DATA): dict,
+    vol.Optional(CONF_NOTIFIERS, default=list): vol.All(cv.ensure_list, [cv.string]),
+}
+
+ALERT_SCHEMA = vol.Schema(STORAGE_FIELDS)
 
 CONFIG_SCHEMA = vol.Schema(
     {DOMAIN: cv.schema_with_slug_keys(ALERT_SCHEMA)}, extra=vol.ALLOW_EXTRA
 )
+RELOAD_SERVICE_SCHEMA = vol.Schema({})
+
+
+class AlertStore(Store):
+    """Store entity registry data."""
 
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Alert component."""
+
     component = EntityComponent[Alert](LOGGER, DOMAIN, hass)
 
-    entities: list[Alert] = []
-
-    for object_id, cfg in config[DOMAIN].items():
-        if not cfg:
-            cfg = {}
-
-        name = cfg[CONF_NAME]
-        watched_entity_id = cfg[CONF_ENTITY_ID]
-        alert_state = cfg[CONF_STATE]
-        repeat = cfg[CONF_REPEAT]
-        skip_first = cfg[CONF_SKIP_FIRST]
-        message_template = cfg.get(CONF_ALERT_MESSAGE)
-        done_message_template = cfg.get(CONF_DONE_MESSAGE)
-        notifiers = cfg[CONF_NOTIFIERS]
-        can_ack = cfg[CONF_CAN_ACK]
-        title_template = cfg.get(CONF_TITLE)
-        data = cfg.get(CONF_DATA)
-
-        entities.append(
-            Alert(
-                hass,
-                object_id,
-                name,
-                watched_entity_id,
-                alert_state,
-                repeat,
-                skip_first,
-                message_template,
-                done_message_template,
-                notifiers,
-                can_ack,
-                title_template,
-                data,
-            )
+    id_manager = collection.IDManager()
+
+    yaml_collection = collection.YamlCollection(
+        logging.getLogger(f"{__name__}.yaml_collection"), id_manager
+    )
+    collection.sync_entity_lifecycle(
+        hass, DOMAIN, DOMAIN, component, yaml_collection, Alert
+    )
+
+    storage_collection = AlertStorageCollection(
```

Comment:
I don't think we should add storage collection support to this integration as our long term plan is to deprecate this integration.

See https://github.com/home-assistant/core/pull/100525#issuecomment-1785204128

Reply:
Okay, that is clear. Is there any acceptable approach to allow reloading the yaml?

Reply:
Maybe, if it doesn't add anything but a service to reload the YAML. But I think we're wasting time when working on this integration as we want to steer users to automations instead.